internal/controller/ovnk_ipfix_collector.go

```diff
@@ -0,0 +1,291 @@
+func parsePort(decodedValue []byte) (uint16, error) {
+	if len(decodedValue) < 2 {
+		return 0, errors.New("insufficient data to parse port")
+	}
+	return binary.BigEndian.Uint16(decodedValue), nil
+}
+
+// parseProtocol converts a byte slice into a protocol string based on IANA protocol numbers.
+// Returns an error if the slice is not the correct size or the protocol is unknown.
+func parseProtocol(decodedValue []byte) (string, error) {
+	if len(decodedValue) < 1 {
+		return "", errors.New("insufficient data to parse protocol")
+	}
+
+	protocol := uint8(decodedValue[0])
+	switch protocol {
+	case 1:
+		return "icmpt", nil
+	case 6:
+		return "tcp", nil
+	case 17:
+		return "udp", nil
+	case 132:
+		return "sctp", nil
+	default:
+		return "", errors.New("unknown protocol")
+	}
+}
+
+// parseIPVersion converts a byte slice into an IP version string (e.g., "ipv4" or "ipv6").
+// Returns an error if the slice is not the correct size or the IP version is unknown.
+func parseIPVersion(decodedValue []byte) (string, error) {
+	if len(decodedValue) < 1 {
+		return "", errors.New("insufficient data to parse IP version")
+	}
+
+	ipVersion := uint8(decodedValue[0])
+	switch ipVersion {
+	case 4:
+		return "ipv4", nil
+	case 6:
+		return "ipv6", nil
+	default:
+		return "", errors.New("unknown IP version")
+	}
+}
+
+// processDataRecord processes a single data record and converts it into an OVNFlow.
+// If any parsing step fails, it returns an error and skips the record.
+func processDataRecord(dataRecord netflows.DataRecord, exportTime uint32) (OVNKFlow, error) {
+	ovnkFlow := OVNKFlow{}
+	for _, field := range dataRecord.Values {
+		switch field.Type {
+		case 8: // sourceIPv4Address
+			sourceIP, err := parseIPv4Address(field.Value.([]byte))
+			if err != nil {
+				return OVNKFlow{}, err
+			}
+			ovnkFlow.SourceIP = sourceIP
+		case 12: // destinationIPv4Address
+			destinationIP, err := parseIPv4Address(field.Value.([]byte))
+			if err != nil {
+				return OVNKFlow{}, err
+			}
+			ovnkFlow.DestinationIP = destinationIP
+		case 7: // sourcePort
+			sourcePort, err := parsePort(field.Value.([]byte))
+			if err != nil {
+				return OVNKFlow{}, err
+			}
+			ovnkFlow.SourcePort = sourcePort
+		case 11: // destinationPort
+			destinationPort, err := parsePort(field.Value.([]byte))
+			if err != nil {
+				return OVNKFlow{}, err
+			}
+			ovnkFlow.DestinationPort = destinationPort
+		case 4: // protocol
+			protocol, err := parseProtocol(field.Value.([]byte))
+			if err != nil {
+				return OVNKFlow{}, err
+			}
+			ovnkFlow.Protocol = protocol
+		case 60: // ipVersion
+			ipVersion, err := parseIPVersion(field.Value.([]byte))
+			if err != nil {
+				return OVNKFlow{}, err
+			}
+			ovnkFlow.IPVersion = ipVersion
+		case 158: // start time delta
+			startTimeDelta := binary.BigEndian.Uint32(field.Value.([]byte))
+			ovnkFlow.StartTimestamp = timestamppb.New(time.Unix(int64(exportTime), 0).Add(-time.Duration(startTimeDelta) * time.Second))
+		default:
```

Comment:
Also record the flow end timestamp as we should update the API to send that for every flow at some point in the future.
```suggestion
		case 159: // flowEndDeltaMicroseconds
			// ...
		default:
```